supported-platforms.md

```diff
@@ -0,0 +1,20 @@
+# Supported Platforms
```

Comment:
Do we want to put this into `/release-engineering`? And link it from the `README.md` there?

Reply:
So I was thinking about that, but I think as this document continues to mature, it actually is generally applicable to sig-release, similar to how the release-blocking / release-informing doc is at root of repo.